components/cli/pkg/commands/project/init.go

```diff
@@ -82,12 +88,61 @@ public function run(cellery:ImageName iName, map<cellery:ImageName> instances, b
     return <@untainted> cellery:createInstance(helloCell, iName, instances, startDependencies, shareDependencies);
 }
 `
+const testImports = `
+import ballerina/test;
+`
+const TestTemplate = `
+import ballerina/test;
+import ballerina/io;
+import celleryio/cellery;
+
+# Handle creation of instances for running tests
+@test:BeforeSuite
+function setup() {
+    cellery:InstanceState[]|error? result = runInstances(conf);
+    cellery:Reference cellEndpoints = <cellery:Reference>cellery:getCellEndpoints(<@untainted> instanceList);
+}
+
+# Tests inserting order from an external cell by calling the pet-be gateway
+@test:Config {}
+function testFunction() {
+    io:println("I'm in test function!");
+    test:assertTrue(true, msg = "Failed!");
+}
 
-// RunInit initializes a cellery project.
-func RunInit(cli cli.Cli, projectName string) error {
+# Handle deletion of instances for running tests
+@test:AfterSuite
+public function cleanUp() {
+    error? err = cellery:stopInstances(instanceList);
+}
+`
+const projStructure = `
+New project structure:
+
+%s/
+├── Ballerina.toml
+└── src
+    └── %s
+        ├── %s
+        └── tests
+            └── %s
+`
+
+func RunInit(projectName string, testArg string) {
+	if testArg == "" {
+		initProject(projectName)
+	} else {
+		err := initTest(projectName)
+		if err != nil {
+			util.ExitWithErrorMessage("error occurred while initializing tests", err)
```

Comment:
Return the error instead of exiting, in all places.

Reply:
Fixed